server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainMappingsRoutesTest.java

```diff
@@ -0,0 +1,387 @@
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+class DomainMappingsRoutesTest {
+    private RecipientRewriteTable recipientRewriteTable;
+    private WebAdminServer webAdminServer;
+
+    private void createServer(DomainMappingsRoutes domainMappingsRoutes) throws Exception {
+        webAdminServer = WebAdminUtils.createWebAdminServer(new DefaultMetricFactory(), domainMappingsRoutes);
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+                .setBasePath("/domainMappings")
+                .log(LogDetail.METHOD)
+                .build();
+    }
+
+    @BeforeEach
+    void setUp() throws Exception {
+        recipientRewriteTable = spy(new MemoryRecipientRewriteTable());
+
+        createServer(new DomainMappingsRoutes(recipientRewriteTable, new JsonTransformer()));
+    }
+
+    @AfterEach
+    void tearDown() {
+        webAdminServer.destroy();
+    }
+
+    @Nested
+    class NormalBehaviour {
+
+        @Test
+        void addDomainMappingShouldRespondWithNoContent() {
+            given()
+                .body("to.com")
+            .when()
+                .put("from.com")
+            .then()
+                .statusCode(HttpStatus.NO_CONTENT_204)
+                .body(isEmptyString());
+        }
+
+        @Test
+        void getDomainMappings() throws RecipientRewriteTableException {
+            String alias1 = "to_1.com";
+            String alias2 = "to_2.com";
+            String alias3 = "to_3.com";
+
+            Domain expectedDomain = Domain.of("abc.com");
+            MappingSource mappingSource = MappingSource.fromDomain(expectedDomain);
+
+            recipientRewriteTable.addAliasDomainMapping(mappingSource, Domain.of(alias1));
+            recipientRewriteTable.addAliasDomainMapping(mappingSource, Domain.of(alias2));
+            recipientRewriteTable.addAliasDomainMapping(mappingSource, Domain.of(alias3));
+
+            Map<String, List<String>> map =
+                    when()
+                        .get()
+                    .then()
+                        .contentType(ContentType.JSON)
+                        .statusCode(HttpStatus.OK_200)
+                    .extract()
+                        .body()
+                        .jsonPath()
+                        .getMap(".");
+
+            assertThat(map)
+                    .containsOnly(entry(expectedDomain.name(), ImmutableList.of(alias1, alias2, alias3)));
+        }
+
+        @Test
+        void getDomainMappingsEmptyMappingsAreFilteredOut() throws RecipientRewriteTableException {
```

Comment:
Here I would suggest `getDomainMappingShouldFilterOutEmptyMappings`

BTW semantic  change from passive form into active form is interesting, as the test name implies we know where the action is taken ;-).